arlas-tagger-core/src/main/java/io/arlas/tagger/service/ManagedKafkaConsumers.java

```diff
@@ -23,33 +23,44 @@
 import io.arlas.tagger.kafka.TagKafkaProducer;
 import io.dropwizard.lifecycle.Managed;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class ManagedKafkaConsumers implements Managed {
     private ArlasTaggerConfiguration configuration;
     private TagRefService tagRefService;
-    private TagExecService tagExecService;
+    private List<TagExecService> tagExecServices;
+    private UpdateServices updateServices;
 
     public ManagedKafkaConsumers(ArlasTaggerConfiguration configuration, TagKafkaProducer tagKafkaProducer, UpdateServices updateServices) {
         this.configuration = configuration;
         this.tagRefService = new TagRefService(configuration,
                 configuration.kafkaConfiguration.tagRefLogTopic,
                 configuration.kafkaConfiguration.tagRefLogConsumerGroupId,
                 tagKafkaProducer, updateServices);
-        this.tagExecService = new TagExecService(configuration,
-                configuration.kafkaConfiguration.executeTagsTopic,
-                configuration.kafkaConfiguration.executeTagsConsumerGroupId,
-                updateServices);
+
+        this.tagExecServices = new ArrayList<>();
+        this.updateServices = updateServices;
     }
 
     @Override
     public void start() throws Exception {
         new Thread(tagRefService).start();
-        new Thread(tagExecService).start();
-
+        for (int i=0; i < configuration.kafkaConfiguration.nbTagExec; i++){
```

Comment:
You may also use an ExecutorService, see https://www.baeldung.com/thread-pool-java-and-guava in order to hide the Threads